### PR TITLE
Fix TRAMP regression

### LINKED ISSUE #29: no difference in mismatch files
Hi,

all files are marked as different (red). For a moment I though that it was some highlighting issue in my configuration, but if I try to customize one of the file names in red (which I know are otherwise identical files) then I get ztreep-diff-model-diff-face, which I suppose it really means that ztree thinks those two files are different.

My problem is so basic that it should be something wrong in my configuration. Any hint for anything I should be looking for? 

Thanks,
 Rodrigo

### PR DESCRIPTION
**Pull request by Silex**
Will fix #29.

Don't merge it yet I'm trying to fix some FIXMEs (detection of different hosts, etc)

### CODE REVIEW

**Comment by Silex** [comment]
@fourier: okay, you can merge

**Comment by fourier** [comment]
it looks like your changes started to be non-trivial. Could you please sign the FSF papers? It will take some time. Please read here: https://github.com/fourier/ztree/#copyright-issues

**Comment by Silex** [comment]
Yawn. Ok I hope at least the FSF papers will work for all GNU projects, I'll keep you updated.

**Comment by fourier** [comment]
Typically you specify the project to work on. I specified "Emacs" and it covers GNU ELPA as well as I understood.

**Comment by Silex** [comment]
I wrote that if possible I'd like it to include GNU* in order to avoid having to resent those :-)

**Comment by Silex** [comment]
Papers sent.

**Comment by fourier** [comment]
Thanks let me know when you get the response from FSF and we can start accepting changes.

**Comment by Silex** [comment]
Hey, I just noticed this:

https://github.com/fourier/ztree/blob/master/ztree-diff-model.el#L152

Do you agree the `-modef` part is useless? shall I remove it? or was it a typo and it should be `-model`?

**Comment by Silex** [comment]
@fourier: ping

**Comment by Silex** [comment]
@fourier: I received the documents!

**Comment by fourier** [comment]
Perfect, thanks. I'll review the patch later this week and test myself with remote connections.

**Comment by Silex** [comment]
@ramestica: mentionning you here so you know when the PR is merged.

File: ztree-diff-model.el [hunk 1]
```diff
@@ -149,19 +149,23 @@ RIGHT if only on the right side."
       file
     (tramp-file-name-localname (tramp-dissect-file-name file))))
 
-(defun ztree-diff-modef-quotify-string (x)
+(defun ztree-diff-quotify-string (x)
```
Comment (fourier) [inline ztree-diff-model.el]: Move to ztree-util, it is a general functon and could be reused

File: ztree-diff-model.el [hunk 2]
```diff
@@ -149,19 +149,23 @@ RIGHT if only on the right side."
...
   "Surround string X with quotes."
   (concat "\"" x "\""))
 
+(defun ztree-same-host-p (file1 file2)
```
Comment (fourier) [inline ztree-diff-model.el]: move to ztree-util, it is a general function and could possibly be reused

File: ztree-diff-model.el [hunk 3]
```diff
@@ -149,19 +149,23 @@ RIGHT if only on the right side."
...
+  "Make an error if FILE1 and FILE2 are not on the same host."
+  (let ((file1-remote (file-remote-p file1))
+        (file2-remote (file-remote-p file2)))
+    (and (equal file1-remote file2-remote) (string-equal file1-remote file2-remote))))
```
Comment (fourier) [inline ztree-diff-model.el]: use string= or string-equal

File: ztree-diff-model.el [hunk 4]
```diff
@@ -259,14 +263,14 @@ setting status from the NODE, unless they have an ignore status"
...
+                      (not
                        (or (eql status 'ignore)
                            (eql (ztree-diff-node-different child) 'ignore))))
               (setf (ztree-diff-node-different child) status)
               (ztree-diff-node-update-diff-from-parent child)))
             children)))
-        
-    
+
+
```
Comment (fourier) [inline ztree-diff-model.el]: Whitespace only change, remove it

File: ztree-diff-model.el [hunk 5]
```diff
@@ -387,7 +391,7 @@ with dot etc)."
   (setf ztree-diff-model-ignore-fun ignore-p))
 
 (defun ztree-diff-model-set-progress-fun (progess-fun)
-  (setf ztree-diff-model-progress-fun progess-fun)) 
+  (setf ztree-diff-model-progress-fun progess-fun))
```
Comment (fourier) [inline ztree-diff-model.el]: Whitespace only change, remove it

File: ztree-diff.el [hunk 6]
```diff
@@ -547,7 +549,7 @@ Argument DIR2 right directory."
     (ztree-diff-update-wait-message (concat "Comparing " dir1 " and " dir2 " ..."))
     (ztree-diff-node-recreate model)
     (message "Done.")
-    
+
```
Comment (fourier) [inline ztree-diff.el]: Whitespace only change, remove it

File: ztree-dir.el [hunk 7]
```diff
@@ -28,7 +28,7 @@
 ;;; Commentary:
 ;;
 ;; Add the following to your .emacs file:
-;; 
+;;
```
Comment (fourier) [inline ztree-dir.el]: Whitespace only change, remove it

File: ztree-dir.el [hunk 8]
```diff
@@ -118,8 +118,8 @@ One could add own filters in the following way:
   "Determines if the file with FILENAME should be visible."
   (let ((name (ztree-file-short-name filename)))
     (and (not (or (string= name ".") (string= name "..")))
-         (or 
-          ztree-dir-show-filtered-files 
+         (or
+          ztree-dir-show-filtered-files
```
Comment (fourier) [inline ztree-dir.el]: Whitespace only change, remove it

File: ztree-dir.el [hunk 9]
```diff
@@ -133,7 +133,7 @@ Otherwise, the ztree window is used to find the file."
            (find-file-other-window node))
           (hard
            (save-selected-window (find-file-other-window node)))
-          (t 
+          (t
```
Comment (fourier) [inline ztree-dir.el]: Whitespace only change, remove it

File: ztree-util.el [hunk 10]
```diff
@@ -3,7 +3,7 @@
 ;; Copyright (C) 2013-2016  Free Software Foundation, Inc.
 ;;
 ;; Author: Alexey Veretennikov <alexey.veretennikov@gmail.com>
-;; 
+;;
```
Comment (fourier) [inline ztree-util.el]: Whitespace only change, remove it

**Comment by fourier** [comment]
@Silex I've also tried to test what you've done.
I've created 2 directories on a remote server with almost identical contents (one file is different and one directory is missing), but got all-red results:

screen shot 2016-09-19 at 21 01 42

It looks like diff was not running on a remote server.
I've run the ztree-diff with Emacs 24.5.1 started with the -Q key to ensure no other packages involved.

**Comment by Silex** [comment]
Alright, I did the requested changes.

About your test, you need to show more information... as you can see on my screenshot, it works... (emacs -Q, version 25.1.50.2 but it worked fine on 24.5.1 when I tested at work).

screenshot from 2016-09-19 21-36-17

How did you connect to the remote server? what TRAMP protocol did you use?

**Comment by fourier** [comment]
I've used tramp ssh.

**Comment by Silex** [comment]
@fourier: once connected to a TRAMP host, try <kbd>M-x eval-expression (shell-command-to-string "ls") RET</kbd>

It should print the remote files.

**Comment by Silex** [comment]
@ramestica: can you test this PR and confirm the fix works for you?

**Comment by Silex** [comment]
@fourier: were you able to test again?

**Comment by fourier** [comment]
No I was not able to test. Waiting for response of @ramestica.

**Comment by ramestica** [comment]
present PR works good to me.

**Comment by fourier** [comment]
Thanks!

**Comment by Silex** [comment]
:+1:

**Pull request merged**